course-v2/assets/js/video_download_popup.js

```diff
@@ -1,16 +1,22 @@
 export const initVideoDownloadPopup = () => {
-  const downloadIcon = document.querySelector(".video-download-icons")
-  const popup = document.querySelector(".video-tab-download-popup")
-  if (downloadIcon) {
-    downloadIcon.addEventListener("click", event => {
-      event.stopPropagation()
-      if (popup) popup.classList.toggle("hidden")
+  const downloadIcons = document.querySelectorAll(".video-download-icons")
+  const popups = document.querySelectorAll(".video-tab-download-popup")
+
+  if (downloadIcons.length > 0) {
+    downloadIcons.forEach((downloadIcon, index) => {
+      const popup = popups[index]
+      downloadIcon.addEventListener("click", event => {
+        event.stopPropagation()
+        if (popup) popup.classList.toggle("hidden")
```

Comment:
In contrast to my comment below, you do need the `if (popup)` here.

**BTW:** could you convert this file to typescript?